pkg/git/git.go

```diff
@@ -0,0 +1,70 @@
+package git
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/codefresh-io/cli-v2/pkg/log"
+)
+
+
+var (
+	requiredGitHubScopes = []string{ "repo", "admin:repo_hook" }
+)
+
+
+func VerifyToken(ctx context.Context, provider string, token string) (bool, error) {
+	providerToVerifier := map[string]func(context.Context, string)(bool, error){}
+
+	providerToVerifier["github"] = verifyGitHubTokenScope
+	providerToVerifier["gitlab"] = verifyGitLabTokenScope
+
+	verifier := providerToVerifier[provider]
+
+	return verifier(ctx, token)
+}
+
+func verifyGitHubTokenScope(ctx context.Context, token string) (bool, error) {
+	log.G(ctx).Info("Verifing your git token")
+
+	req, _ := http.NewRequestWithContext(ctx, "HEAD", "https://api.github.com/", nil)
+	req.Header.Set("Authorization", fmt.Sprintf("token %s", token))
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return false, err
+	}
+
+	defer resp.Body.Close()
+
+	rawScopes := resp.Header["X-Oauth-Scopes"]
+	var scopes []string
+	if len(rawScopes) > 0 {
+		scopes = strings.Split(rawScopes[0], ", ")
+	}
+	var repo bool
+	var adminRepoHook bool
+
+	for _, scope := range scopes {
+		if scope == requiredGitHubScopes[0] {
```

Comment:
This is not what I expected..
you have an array of scopes you need to verify. tomorrow I will add additional 10 scopes.